### PR TITLE
Issue #30 -addresses false positive in cycle detection

### PR DESCRIPTION
**Pull request by mbpreble**
This updates the path comparison used in cycle detection to ignore the final component of the path for references which have been seen before. This fixes an edge case where false positives were reported.

There's a unit test for `jsan.stringify` verifying the change, which fails when it is reverted. Existing tests pass.

### CODE REVIEW

File: lib/cycle.js [hunk 1]
```diff
@@ -111,7 +111,14 @@ exports.decycle = function decycle(object, options, replacer) {
           if (noCircularOption && withRefs) {
             return {$jsan: foundPath};
           }
-          if (path.indexOf(foundPath) === 0) {
+          
+          // This is only a true circular reference if the parent path is inside of foundPath
+          // drop the last component of the current path and check if it starts with foundPath
+          var pathElements = path.split('.');
+          var parentPathElements = pathElements.slice(0, pathElements.length - 1)
```
Comment (kolodny) [inline lib/cycle.js]: nit: `pathElements.slice(0, -1)`

**Review by kolodny — CHANGES_REQUESTED** [review]
Looks good, just one small nit

File: lib/cycle.js (hunk 1 above)
Comment (mbpreble) [reply lib/cycle.js]: Thanks, should be able to turn this into a one-liner if I'm not doing that.

Comment (mbpreble) [reply lib/cycle.js]: Cleaned this up a bit.

**Comment by kolodny** [comment]
Thanks!
